File: spanish-legacy/narcisistas/Coding_Challenge_2.java
public class Coding_Challenge_2{
    public static void Narcisistas(int n){
        for(int i=0;i<=n;i++){
            String j = Integer.toString(i);
            int p = j.length();
            int s = 0;
            for(int k=0;k<p;k++){
                char ch = j.charAt(k);
                int d = Character.getNumericValue(ch);
                s += Math.pow(d,p);
            }
            if(s==i){
                System.out.println(s);
            }
        }
    }
}

File: spanish-legacy/R/coding_challenge_2.R
narcisistas = function(n){
  for(i in 0:n){
    j = as.character(i);
    p = nchar(j);
    s = 0;
    for(k in 1:p){
      ch = substr(j,k,k);
      d = strtoi(ch);
      s = s+d^p;
    }
    if(s==i){
      print(i)
    }
  }
}
